src/codegen/generate.hpp

```diff
@@ -111,6 +112,13 @@ class NasmGenerator_x86_64
     }, location);
   }
 
+  void emitBinaryOp(const char *op, const auto &tgt, const auto &src) {
```

Comment:
I prefer to force having `scopes::Register` here and transform `GeneralPurposeRegister` with `getProperRegisterFromID64` because of possible size issues. In fact I was expecting to use GeneralPurposeRegisters in the ast namespace and real registers in codegen. Because in ast we want to specify registers but we don't want to have to care of the data size

Reply:
You're right! lemme do that